pyccel/codegen/printing/fcode.py

```diff
@@ -603,13 +602,15 @@ def _print_SumFunction(self, expr):
     def _print_PythonLen(self, expr):
         var = expr.arg
         idx = 1 if var.order == 'F' else var.rank
+        prec = iso_c_binding["integer"][expr.precision]
+
         dtype = var.dtype
         if dtype is NativeString():
             return 'len({})'.format(self._print(var))
         elif var.rank == 1:
-            return 'size({})'.format(self._print(var))
+            return 'size({}, kind={})'.format(self._print(var), prec)
         else:
-            return 'size({},{})'.format(self._print(var), self._print(idx))
+            return 'size({},{},{})'.format(self._print(var), self._print(idx), prec)
```

Comment:
In order to be consistent with line 611, shouldn't we pass `prec` through the `kind` keyword argument?

Reply:
the keyword needs to be specified in line 611 otherwise it's not gonna work, the other line I prefere not because we use it alot and it will make the code less readable.